cmd/readOnlyLib/main.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"C"
+
+	"github.com/ethereum/go-ethereum/internal/debug"
+)
+import (
+	"context"
+	"os"
+	"unsafe"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var stack *node.Node
+
+//export open_database
+func open_database(datadir *C.char) C.int {
+	glogger := log.NewGlogHandler(log.StreamHandler(os.Stderr, log.TerminalFormat(false)))
+	glogger.Verbosity(0)
+	log.Root().SetHandler(glogger)
```

Comment:
Ideally the caller can pass in a Callback here for handling logging.

This is the best pointer I could find for how to do it:

Start reading at:
```Calling C function pointers is currently not supported, however you can declare Go variables which hold C function pointers and pass them back and forth between Go and C. C code may call function pointers received from Go.```

here https://pkg.go.dev/cmd/cgo#hdr-Go_references_to_C.

Let me know if it works!